qiskit/utils/__init__.py

```diff
@@ -70,7 +71,7 @@
 
 from . import optionals
 
-from .parallel import parallel_map
+from .parallel import parallel_map, should_run_in_parallel
```

Comment:
For backport I was thinking we didn't expose this as part of the public api. Right now it's really only needed for internal decision making.

Reply:
Does it work for you if I leave this PR like this for 1.2, then suppress the documentation and release note of the new function during the backport to 1.1 stable?

Reply:
If you don't mind doing a manual backport that's fine with me. Personally I'd make the diff on main adding the docs and release note because that seems less error prone imo. But doing it that way works fine

Reply:
Let me do that, then - it'll be clearer to see what we did when if we look back in a few months' time.

Reply:
Done in 5fc08aea3c.